src/detail/flare.cc

```diff
@@ -1,69 +1,93 @@
 #include "broker/detail/flare.hh"
 
 #include <errno.h>
-#include <fcntl.h>
-#include <poll.h>
-#include <unistd.h>
 
-#include <exception>
 #include <algorithm>
+#include <exception>
 
+#include "broker/config.hh"
+#include "broker/detail/assert.hh"
 #include "broker/logger.hh"
 
-namespace broker {
-namespace detail {
+#ifdef BROKER_WINDOWS
 
-namespace {
+#include <Winsock2.h>
+template <class... Ts>
+auto poll(Ts... xs) {
+  return WSAPoll(xs...);
+}
 
-constexpr size_t stack_buffer_size = 256;
+#define PIPE_WRITE(fd, buf, num_bytes) ::send(fd, buf, num_bytes, 0)
 
-} // namespace <anonymous>
+#define PIPE_READ(fd, buf, num_bytes) ::recv(fd, buf, num_bytes, 0)
 
-flare::flare() {
-  if (::pipe(fds_) == -1) {
-    BROKER_ERROR("failed to create flare pipe");
-    std::terminate();
-  }
+bool try_again_later() {
+  int code = WSAGetLastError();
+  return code == WSAEWOULDBLOCK;
+}
 
-  int res;
+#else // BROKER_WINDOWS
 
-  res = ::fcntl(fds_[0], F_SETFD, ::fcntl(fds_[0], F_GETFD) | FD_CLOEXEC);
+#include <errno.h>
+#include <fcntl.h>
+#include <poll.h>
+#include <unistd.h>
+
+#define PIPE_WRITE ::write
+
+#define PIPE_READ ::read
+
+bool try_again_later() {
```

Comment:
Expected `try_again_later` and `poll` (for the windows section) to be in unnamed namespace or else `static`.